python/tk_desktop/setup_project.py

```diff
@@ -62,10 +69,57 @@ def do_setup(self):
                                        "and delete all pipeline configurations for your project.")
             ret = error_dialog.exec_()
 
+        except TankUserPermissionsError, e:
+            error_dialog = ErrorDialog("Toolkit Setup Error",
+                                       "You do not have sufficient permissions in Shotgun to setup Toolkit for "
+                                       "project '%s'.\n\nContact a site administrator for assistance." %
+                                        self.project["name"]
+            )
+            ret = error_dialog.exec_()
+
         finally:
             if is_on_top: 
                 self._set_top_window_on_top(True)
 
+    def show_help_popup(self):
+        """
+        Display a help screen
+        """
+        # Not sure what to put here.
+        pass
+
+    def _validate_user_permissions(self):
+        """
+        Attempt to modify the Project's tank_name field to determine whether
+        the current user has sufficient permission to setup the Project's
+        pipeline configuration.
+        """
+        try:
+            # Try to update the Project's tank_name value in SG to test
+            # whether current user has sufficient permission to setup
+            # Toolkit for a project.
+            engine = sgtk.platform.current_engine()
+            sg_project = engine.shotgun.find_one(
+                "Project", [["id", "is", self.project["id"]]], ["tank_name"]
+            )
+            engine.shotgun.update(
+                "Project", self.project["id"], {"tank_name": "foobar"}
+            )
+            engine.shotgun.update(
+                "Project", self.project["id"], {"tank_name": sg_project["tank_name"]}
+            )
+        except Exception, e:
```

Comment:
Might be worth catching an shotgun_api3.Fault instead to announce in advance the intent of the catch.